build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -24,15 +24,27 @@ IMG_TOOL_PATH=${TOOL_PATH}/image_tool
 OS_PATH=${TOP_PATH}/os
 CONFIG=${OS_PATH}/.config
 source ${CONFIG}
+APP_NUM=0
 
 function pre_download()
 {
 	source ${TOP_PATH}/os/.bininfo
 	cp -p ${BIN_PATH}/km0_boot_all.bin ${IMG_TOOL_PATH}/km0_boot_all.bin
 	cp -p ${BIN_PATH}/km4_boot_all.bin ${IMG_TOOL_PATH}/km4_boot_all.bin
 	cp -p ${BIN_PATH}/${KERNEL_BIN_NAME} ${IMG_TOOL_PATH}/${KERNEL_BIN_NAME}
-	cp -p ${BIN_PATH}/micom ${IMG_TOOL_PATH}/micom
-	cp -p ${BIN_PATH}/wifi ${IMG_TOOL_PATH}/wifi
+	if test -f "${BIN_PATH}/micom"; then
+		APP_NUM=$(($APP_NUM+1))
+		cp -p ${BIN_PATH}/micom ${IMG_TOOL_PATH}/micom
+	fi
+	if test -f "${BIN_PATH}/wifi"; then
+		APP_NUM=$(($APP_NUM+1))
+		cp -p ${BIN_PATH}/wifi ${IMG_TOOL_PATH}/wifi
+	fi
+	if [ ${APP_NUM} -eq 0 ]; then
+		echo "No User Binary."
+		post_download
+		exit 1
+	fi
```

Comment:
I think this will cause the download to fail on configs other than loadable. Please add a condition to check if it is loadable config before checking the APP_NUM.

Reply:
Added. Thanks.